core-api/src/main/java/com/optimizely/ab/bucketing/DecisionService.java

```diff
@@ -212,7 +218,13 @@ public DecisionService(@Nonnull Bucketer bucketer,
         int rolloutRulesLength = rollout.getExperiments().size();
         String bucketingId = userId;
         if (filteredAttributes.containsKey(ControlAttribute.BUCKETING_ATTRIBUTE.toString())) {
-            bucketingId = filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString());
+            if (String.class.isInstance(filteredAttributes.get(ControlAttribute.BUCKETING_ATTRIBUTE.toString()))) {
```

Comment:
This logic is repeated from up above. Can you consolidate into a helper function?